### ISSUE TITLE
Replace index.mds with symbolic links to latest document version

### ISSUE THREAD

**Issue by m-aciek** [semver/semver.org#175]
For now there is redundancy in the repository. `index.md` and `spec/v2.0.0.md` are intended to be equal for main directory and translations.

It causes confusion and unsyncing of files because of one-file edits in translations (b43f3ef6, 4fa99c5ea, aafc4b32).

Symbolic links would make things simple. Jekyll handles them well on filesystems supporting symbolic links.

What do you think?

**Comment by m-aciek** [comment]
@Haacked do you have opinion on that? I [had started to migrate the files](https://github.com/m-aciek/semver.org/commits/gh-pages-symlinks) but it looks the are too many files to resolve conflicts for them at once. I would start to make PRs language by language.

I would divide it into smaller pieces:
- [ ] change instruction for adding a translation in Readme and create symbolic link for (English) index (https://github.com/semver/semver.org/commit/7a3ded02dd1a1732afe121452c16df54ad26b50d & https://github.com/semver/semver.org/commit/133fbc004308d542f179f9e899e7183b29dc7b99)
- [ ] ar (https://github.com/semver/semver.org/commit/00aed5f49b6bd3afe1c40af408ec984120726a43)
- [ ] cs (https://github.com/semver/semver.org/commit/d48b8c967789d0c9f32aa835bcf86e3858d46386)
- [ ] de (https://github.com/semver/semver.org/commit/d01746a5246e2d335018d419aca9a52f47d3a26e)
- [ ] es (https://github.com/semver/semver.org/commit/f071ac961135c6c5fb118502b12b86829e80b35a)
- [ ] fa (https://github.com/semver/semver.org/commit/0e7633ee8dd282471eb4eeb02f80e1176f1c2e00)
- [ ] fr (https://github.com/semver/semver.org/commit/ea0790d042d9b57d0df45e7db098ba9b162bfa35)
- [ ] he (https://github.com/semver/semver.org/commit/a18726588ba4e107e975a4cb737a425287c4e903)
- [ ] id (https://github.com/semver/semver.org/commit/3e09151d7b87fe3c1607c5b3cca3f6f3bacb98e5)
- [ ] it (https://github.com/semver/semver.org/commit/c113ce71e7190f54313629f79a88716036580f89)
- [ ] ja
- [ ] ka
- [ ] ko
- [ ] pl (https://github.com/semver/semver.org/commit/c89f3e9f37bd8725e7bd0850daf3fa6eac2561a8 & https://github.com/semver/semver.org/commit/fb44b4c2e740bcbcdc00ec041dbd7c49062c44c1)
- [ ] pt-BR
- [ ] ru
- [ ] sk
- [ ] sl
- [ ] sv
- [ ] tr
- [ ] zh-CN
- [ ] zh-TW

**Comment by haacked** [comment]
In theory I love this idea, but the maintenance of it might be challenging. Would it be possible to script this in a batch file/shell script so that we can always fix this up easily?

**Comment by damianopetrungaro** [comment]
Yeah, it's possible to be done, maybe making a `Makefile` will be easier to use too.

We are facing the same issue on conventional-commits, probably we'll migrate from jekyll to hugo and we won't need to use symlinks.

I'll probably have some free time this week if it's ok for you I can work on it.

**Comment by m-aciek** [comment]
The script should work well with [git merge-file utility](https://git-scm.com/docs/git-merge-file). I tested it locally and all tests were successful so far.

**Comment by haacked** [comment]
I'm open to the idea. I'm not interested in moving away from Jekyll, so perhaps something I can run locally to fix up the files would work well.

**Comment by grv87** [comment]
I think that custom Jekyll plugin put in `_plugins` dir could duplicate latest specification to index pages either before or after generation.
Then the DRY principle would be respected, and complicated git tricks wouldn't be necessary.

Maybe there is an existing plugin, but I couldn't find any.

**Comment by alexandrtovmach** [comment]
It's really great idea and I'll investigate to it later. Added to https://github.com/semver/semver/issues/573

**Comment by rmjspaz7284** [comment]
> For now there is redundancy in the repository. `index.md` and `spec/v2.0.0.md` are intended to be equal for main directory and translations.
> 
> It causes confusion and unsyncing of files because of one-file edits in translations ([b43f3ef](https://github.com/semver/semver.org/commit/b43f3ef6d5344c7d7ff6948c2f6a34ca76ecbb6f), [4fa99c5](https://github.com/semver/semver.org/commit/4fa99c5eaa1fb40de2a71b63852ff725c92c7b13), [aafc4b3](https://github.com/semver/semver.org/commit/aafc4b3264fe59d287b9326d6363d2b43508223b)).
> 
> Symbolic links would make things simple. Jekyll handles them well on filesystems supporting symbolic links.
> 
> What do you think?

**Comment by LaoshuBaby** [comment]
I really like this idea, it will reduce translator's burden on manual sync latest version with index.md, and will not lead to a situation that some translator forget to write index.md but only translate the v2.0.0.md (was fixed in Vietnamese version)

**Comment by yochem** [comment]
I hadn't noticed this issue yet, but I tried tackling this problem in another way in #448 (at least for the english part). It uses liquid to include `spec/v{ current_version }.md`. Current version is set in the config file.

I think it would be a good idea to review the complete workflow of getting different version of the spec and building it could be revised. It seems that it could be a lot easier. Also, why not include the translations in the spec repository? Because they are not the "original" spec?
